### PR TITLE
Cherrypick/aosp18 11 2021

### PR DESCRIPTION
**Pull request by igordmn**
The new code only in "Fix build" commit, other commits just cherrypicks from aosp.

### CODE REVIEW

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/ScrollableState.kt [hunk 1]
```diff
@@ -144,8 +144,11 @@ private class DefaultScrollableState(val onDelta: (Float) -> Float) : Scrollable
     ): Unit = coroutineScope {
         scrollMutex.mutateWith(scrollScope, scrollPriority) {
             isScrollingState.value = true
-            block()
-            isScrollingState.value = false
+            try {
+                block()
+            } finally {
+                isScrollingState.value = false
```
Comment (olonho) [inline compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/ScrollableState.kt:150]: Why this change?

Comment (igordmn) [reply compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/ScrollableState.kt:150]: It fixes a bug, so I cherry-picked it

**Review by olonho — APPROVED** [review]
